components/capi/include/pluralrules.h

```diff
@@ -0,0 +1,46 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#pragma once
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "provider.h"
+#include "locale.h"
+
+// opaque
+typedef struct PluralRules PluralRules; 
+
+typedef struct {
+    PluralRules* rules;
+    bool success;
+} CreatePluralRulesResult;
+
+typedef enum {
+    PluralRuleType_Cardinal,
+    PluralRuleType_Ordinal
+} PluralRuleType;
+
+typedef enum {
+    PluralCategory_Zero,
+    PluralCategory_One,
+    PluralCategory_Two,
+    PluralCategory_Few,
+    PluralCategory_Many,
+    PluralCategory_Other,
+} PluralCategory;
+
+typedef struct {
+    uint64_t i;
+    size_t v;
+    size_t w;
+    uint64_t f;
+    uint64_t t;
+    size_t c;
+} PluralOperands;
```

Comment:
For the number of operands we have I think this is a good approach. The only other thing that comes to mind would be an enum for the possible operand types, and then a struct with the operand type and a union (yucky...) to hold possible values. This would allow callers to pass in an array of values. That might be something to consider in the future if we expose an API with a large number of options.